### PR TITLE
feat: rm upload-api-proxy ability to route to separate environment audiences

### PR DESCRIPTION
**Pull request by gobengo**
Motivation:
* https://github.com/web3-storage/w3protocol/issues/406
  * tl;dr remove some code/behavior that's not strictly needed and @hugomrdias asked me to remove

### CODE REVIEW

**Comment by gobengo** [comment]
@hugomrdias does this fulfill what you asked for [here](https://github.com/web3-storage/w3protocol/pull/390#pullrequestreview-1275740500)?

**Review by hugomrdias — CHANGES_REQUESTED** [review]

File: packages/access-api/wrangler.toml [hunk 1]
```diff
@@ -60,7 +60,7 @@ unsafe = { bindings = [
 [env.staging]
 name = "w3access-staging"
 workers_dev = true
-vars = { ENV = "staging", DEBUG = "false", DID = "did:web:staging.web3.storage" }
+vars = { ENV = "staging", DEBUG = "false", DID = "did:web:staging.web3.storage", UPLOAD_API_URL = "https://staging.up.web3.storage" }
```
[the remark this answers is not in the document]
Comment (gobengo) [reply packages/access-api/wrangler.toml:63]: like this? https://github.com/web3-storage/w3protocol/pull/407/commits/a0f07b5c993cfb32c218bb7de78695cbeaf2836a

afaict upload-api doesn't have a 'dev' environment so I pointed it to upload-api staging

File: packages/access-api/src/bindings.d.ts [hunk 2]
```diff
@@ -55,8 +54,7 @@ export interface RouteContext {
     validations: Validations
   }
   uploadApi: {
-    production?: URL
-    staging?: URL
+    ucanto?: { url: URL }
```
[the remark this answers is not in the document]
Comment (gobengo) [reply packages/access-api/src/bindings.d.ts]: I did it here, wdyt? https://github.com/web3-storage/w3protocol/pull/407/commits/cc94badfd6f5569ad17aa9649e85d9e54bb3ea94#diff-a7cd4d05e6cba234e750cd72e2c9301b4cecec823d94002c39e928edd73b5b41L56

File: packages/access-api/src/service/upload-api-proxy.js [hunk 3]
```diff
@@ -73,21 +69,16 @@ const uploadApiEnvironments = {
...
+      // use production environment
       ...uploadApiEnvironments.production,
-      ...(uploadApi.production && { url: uploadApi.production }),
-      fetch,
-    }),
-    [uploadApiEnvironments.staging.audience]: createUcantoHttpConnection({
-      ...uploadApiEnvironments.staging,
-      url: uploadApi.staging ?? uploadApiEnvironments.staging.url,
+      // but override uploadApi.url from configuration, if provided
+      ...uploadApi.ucanto,
```
[the remark this answers is not in the document]
Comment (gobengo) [reply packages/access-api/src/service/upload-api-proxy.js]: I removed this default behavior by making the relevant options required and removed this function. https://github.com/web3-storage/w3protocol/pull/407/commits/cc94badfd6f5569ad17aa9649e85d9e54bb3ea94#diff-969dad085be9e4a6140a50e5f0142a0e6f106b12a5c3375f5857006fbe71dab8L74

**Review by hugomrdias — APPROVED** [review]
LGTM

**Pull request merged**